dockerfiles/che-launcher/launcher.sh

```diff
@@ -0,0 +1,430 @@
+  debug "CHE_HOSTNAME              = ${CHE_HOSTNAME}"
+  debug "CHE_DATA_FOLDER           = ${CHE_DATA_FOLDER}"
+  debug "CHE_CONF_FOLDER           = ${CHE_CONF_FOLDER:-not set}"
+  debug "CHE_LOCAL_BINARY          = ${CHE_LOCAL_BINARY:-not set}"
+  debug ""
+  debug "---------------------------------------"
+  debug "---------------------------------------"
+  debug "---------------------------------------"
+}
+
+is_boot2docker() {
+  if uname -r | grep -q 'boot2docker'; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+has_docker_for_windows_ip() {
+  DOCKER_HOST_IP=$(get_docker_host_ip)
+  if [ "${DOCKER_HOST_IP}" = "10.0.75.2" ]; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+is_docker_for_mac() {
+  if uname -r | grep -q 'moby' && ! has_docker_for_windows_ip; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+is_docker_for_windows() {
+  if uname -r | grep -q 'moby' && has_docker_for_windows_ip; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+get_docker_install_type() {
+  if is_boot2docker; then
+    echo "boot2docker"
+  elif is_docker_for_windows; then
+    echo "docker4windows"
+  elif is_docker_for_mac; then
+    echo "docker4mac"
+  else
+    echo "native"
+  fi
+}
+
+get_docker_host_ip() {
+  NETWORK_IF="eth0"
+  if is_boot2docker; then
+    NETWORK_IF="eth1"
+  fi
+
+  docker run --rm --net host \
+            alpine sh -c \
+            "ip a show ${NETWORK_IF}" | \
+            grep 'inet ' | \
+            cut -d/ -f1 | \
+            awk '{ print $2}'
+}
+
+get_che_hostname() {
+  INSTALL_TYPE=$(get_docker_install_type)
+  if [ "${INSTALL_TYPE}" = "boot2docker" ] ||
+     [ "${INSTALL_TYPE}" = "docker4windows" ]; then
+    get_docker_host_ip
+  else
+    echo "localhost"
+  fi
+}
+
+check_docker() {
+  if [ ! -S /var/run/docker.sock ]; then
+    error_exit "Docker socket (/var/run/docker.sock) hasn't be mounted \
+inside the container. Verify the syntax of the \"docker run\" command."
+  fi
+
+  if ! docker ps > /dev/null 2>&1; then
+    output=$(docker ps)
+    error_exit "Error when running \"docker ps\": ${output}"
+  fi
+}
+
+che_container_exist() {
+  if [ "$(docker ps -aq  -f "name=${CHE_SERVER_CONTAINER_NAME}" | wc -l)" = "0" ]; then
+    return 1
+  else
+    return 0
+  fi
+}
+
+che_container_is_running() {
+  if [ "$(docker ps -qa -f "status=running" -f "name=${CHE_SERVER_CONTAINER_NAME}" | wc -l)" = "0" ]; then
+    return 1
+  else
+    return 0
+  fi
+}
+
+che_container_is_stopped() {
+  if [ "$(docker ps -qa -f "status=exited" -f "name=${CHE_SERVER_CONTAINER_NAME}" | wc -l)" = "0" ]; then
+    return 1
+  else
+    return 0
+  fi
+}
+
+wait_until_container_is_running() {
+  CONTAINER_START_TIMEOUT=${1}
+
+  ELAPSED=0
+  until che_container_is_running || [ ${ELAPSED} -eq "${CONTAINER_START_TIMEOUT}" ]; do
+    sleep 1
+    ELAPSED=$((ELAPSED+1))
+  done
+}
+
+server_is_booted() {
+  HTTP_STATUS_CODE=$(curl -I http://"${CHE_HOST_IP}":"${CHE_PORT}"/api/ 2>/dev/null \
```

Comment:
you can also use specific options of curl instead of manual parsing of output 
`curl http://"${CHE_HOST_IP}":"${CHE_PORT}"/api/ -s -o /dev/null --write-out "%{http_code}"`
where 
-s do not print additional info 
-o /dev/null suppress response body
--write-out "%{http_code}"` print status code